### ISSUE TITLE
Plan to support Python 3.11

### ISSUE THREAD

**Issue by mrchoke** [permitio/fastapi_websocket_pubsub#49]
I try to run with Python 3.11 but it can't.

```
Traceback (most recent call last):
  File "/workspace/api/main.py", line 6, in <module>
    from fastapi_websocket_pubsub import PubSubEndpoint
  File "/usr/local/lib/python3.11/site-packages/fastapi_websocket_pubsub/__init__.py", line 1, in <module>
    from .pub_sub_server import PubSubEndpoint
  File "/usr/local/lib/python3.11/site-packages/fastapi_websocket_pubsub/pub_sub_server.py", line 5, in <module>
    from fastapi_websocket_rpc import WebsocketRPCEndpoint
  File "/home/fastapi/.local/lib/python3.11/site-packages/fastapi_websocket_rpc/__init__.py", line 2, in <module>
    from .websocket_rpc_client import WebSocketRpcClient
  File "/home/fastapi/.local/lib/python3.11/site-packages/fastapi_websocket_rpc/websocket_rpc_client.py", line 12, in <module>
    from .rpc_channel import RpcChannel, OnConnectCallback, OnDisconnectCallback
  File "/home/fastapi/.local/lib/python3.11/site-packages/fastapi_websocket_rpc/rpc_channel.py", line 7, in <module>
    from asyncio.coroutines import coroutine
ImportError: cannot import name 'coroutine' from 'asyncio.coroutines' (/usr/local/lib/python3.11/asyncio/coroutines.py)

```

**Comment by orweis** [comment]
Hi :) 

The changes needed are actually in the underlying RPC library.
I've opened a PR there, will merge after a review. 

CC: @asafc

**Comment by mrchoke** [comment]
> Hi :)
> 
> The changes needed are actually in the underlying RPC library. I've opened a [PR](https://github.com/permitio/fastapi_websocket_rpc/pull/18) there, will merge after a review.

Thank you for your response. 
> 
> CC: @asafc


I try to run pytest on **fastapi_websocket_pubsub** after patch **fastapi_websocket_rpc** it not pass **_tests/broadcaster_test.py_** .

```
DEBUG    websockets.client:protocol.py:256 = connection is CONNECTING
DEBUG    websockets.client:client.py:114 > GET /ws1 HTTP/1.1
DEBUG    websockets.client:client.py:116 > Host: localhost:7990
DEBUG    websockets.client:client.py:116 > Upgrade: websocket
DEBUG    websockets.client:client.py:116 > Connection: Upgrade
DEBUG    websockets.client:client.py:116 > Sec-WebSocket-Key: vH4jO8icQ+96dnZFBBolig==
DEBUG    websockets.client:client.py:116 > Sec-WebSocket-Version: 13
DEBUG    websockets.client:client.py:116 > Sec-WebSocket-Extensions: permessage-deflate; client_max_window_bits
DEBUG    websockets.client:client.py:116 > User-Agent: Python/3.11 websockets/10.4
DEBUG    websockets.client:client.py:147 < HTTP/1.1 500 Internal Server Error
DEBUG    websockets.client:client.py:149 < content-type: text/plain; charset=utf-8
DEBUG    websockets.client:client.py:149 < content-length: 21
DEBUG    websockets.client:client.py:149 < connection: close
DEBUG    websockets.client:protocol.py:1429 ! failing connection with code 1006
DEBUG    websockets.client:protocol.py:1334 x half-closing TCP connection
DEBUG    websockets.client:protocol.py:1514 = connection is CLOSED
```

it has 500  Internal Server Error

**Comment by mrchoke** [comment]
fastapi error log

```
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/uvicorn/protocols/websockets/websockets_impl.py", line 238, in run_asgi
    result = await self.app(self.scope, self.asgi_receive, self.asgi_send)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/fastapi/applications.py", line 270, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/starlette/applications.py", line 124, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 149, in __call__
    await self.app(scope, receive, send)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/starlette/routing.py", line 706, in __call__
    await route.handle(scope, receive, send)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/starlette/routing.py", line 341, in handle
    await self.app(scope, receive, send)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/starlette/routing.py", line 82, in app
    await func(session)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/venv/lib/python3.11/site-packages/fastapi/routing.py", line 287, in app
    await dependant.call(**values)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/tests/broadcaster_test.py", line 72, in websocket_rpc_endpoint
    await endpoint.main_loop(websocket)
  File "/Users/mrchoke/Work/python/fastapi_websocket_pubsub/fastapi_websocket_pubsub/pub_sub_server.py", line 137, in main_loop
    done, pending = await asyncio.wait([self.endpoint.main_loop(websocket, client_id=client_id, **kwargs),
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.0/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/tasks.py", line 415, in wait
    raise TypeError("Passing coroutines is forbidden, use tasks explicitly.")
TypeError: Passing coroutines is forbidden, use tasks explicitly.

```

**Comment by orweis** [comment]
Hi @mrchoke - sorry I missed your message.
Maybe your patch didn't work as expected?

I've released the RPC as new version (0.1.24)
And dependent tests here pass successfully:
https://github.com/permitio/fastapi_websocket_pubsub/pull/50 

Will release soon as a new version, after CR.

**Comment by orweis** [comment]
Closed by https://github.com/permitio/fastapi_websocket_pubsub/pull/50
